package.json

```diff
@@ -54,7 +61,11 @@
     "coveragePathIgnorePatterns": [
       "/node_modules/",
       "/Bootstrap/",
+      "/test/",
```

Comment:
Hmm maybe switch this to `Stub` and `CrypterTest` to `CrypterStub` ... test kind of gives the feeling that this is used for TDD

Reply:
Also using `/test/` directories for storing test data shared between different tests (e.g. `Domain/UseCase/CreateValetToken/test/data.ts`.

I think it's useful to have part of the directory structure reserved for tests only (including stubs, fakes or whatchacallits) -- where you keep test utils and data (which are of course exempt from testing themselves) -- so all things used in tests only would go under `/test/` in the appropriate places.

We could also do it Java-style by having a single `test` directory somewhere near the top level which contains tests and test data separately in a replicated directory structure, altho this would be a bit against convention used in other projects here. 

Or maybe there is another solution we can come up with. Anyway the current one feels good enough for me.

Reply:
Test data should be mocked in the spec for a given class - I don't think there's much reuse between two specs of a different class. 

If we're talking about reusable test data then I'm thinking more of E2E testing rather than unit testing - wdyt?

Reply:
I think reusability is unrelated to the kind of test (E2E, unit, integration, whathaveyou). If the data is starting to get duplicated, especially when we are dealing with cross-cutting concerns or related modules then the sensible thing is to extract it to reduce copy-pasting and the volume of code that needs to be changed when we change data types or other popular dependents.

The mentioned `Domain/UseCase/CreateValetToken/test/data.ts` qualifies -- this data is reused across specs that deal with `CreateValetKeyDto`s. Same for `ValetPayload`.

I suppose we could put the data in an arbitrary spec instead and import from there, but this doesn't feel right.

Reply:
I wanted to avoid the situation where we would have to put same efforts into maintenance of tests as of maintenance of code. Having that said your arguments about reusing some structures do make sense - thus I conclude this resolved :)